src/v2/controllers/FormController.js

```diff
@@ -156,7 +161,22 @@ export default Controller.extend({
         .then(this.handleIdxSuccess.bind(this))
         .catch(error => {
           this.showFormErrors(this.formView.model, error);
+        })
+        .then(() => {
+          if (actionPath === 'cancel') {
+            // TODO: remove this line after OKTA-405474
+            if (useInteractionCodeFlow) {
+              this.options.settings.set('useInteractionCodeFlow', true);
+            }
+            this.options.appState.trigger('interactionCanceled', this.constructor);
```

Comment:
why need to pass `this.constructor` ?

Reply:
Because this will be passed as first parameter (`Controller`) to `BaseLoginRouter.render()` 
https://github.com/okta/okta-signin-widget/blob/1bca146103174f2993c2412b3fe7bc7a3e7fd7a8/src/v2/BaseLoginRouter.js#L83
https://github.com/okta/okta-signin-widget/blob/1bca146103174f2993c2412b3fe7bc7a3e7fd7a8/src/v2/BaseLoginRouter.js#L189